streamer/src/nonblocking/quic.rs

```diff
@@ -84,69 +160,17 @@ pub async fn run_server(
-                stats.total_new_connections.fetch_add(1, Ordering::Relaxed);
-                let NewConnection {
-                    connection,
-                    uni_streams,
-                    ..
-                } = new_connection;
-
-                let remote_addr = connection.remote_address();
-
-                let (mut connection_table_l, stake) = {
-                    let staked_nodes = staked_nodes.read().unwrap();
-                    if let Some(stake) = staked_nodes.get(&remote_addr.ip()) {
-                        let stake = *stake;
-                        drop(staked_nodes);
-                        let mut connection_table_l = staked_connection_table.lock().unwrap();
-                        let num_pruned = connection_table_l.prune_oldest(max_staked_connections);
-                        stats.num_evictions.fetch_add(num_pruned, Ordering::Relaxed);
-                        (connection_table_l, stake)
-                    } else {
-                        drop(staked_nodes);
-                        let mut connection_table_l = connection_table.lock().unwrap();
-                        let num_pruned = connection_table_l.prune_oldest(max_unstaked_connections);
-                        stats.num_evictions.fetch_add(num_pruned, Ordering::Relaxed);
-                        (connection_table_l, 0)
-                    }
-                };
-
-                if stake != 0 || max_unstaked_connections > 0 {
-                    if let Some((last_update, stream_exit)) = connection_table_l.try_add_connection(
-                        &remote_addr,
-                        timing::timestamp(),
-                        max_connections_per_ip,
-                    ) {
-                        drop(connection_table_l);
-                        let packet_sender = packet_sender.clone();
-                        let stats = stats.clone();
-                        let connection_table1 = connection_table.clone();
-                        tokio::spawn(handle_connection(
-                            uni_streams,
-                            packet_sender,
-                            remote_addr,
-                            last_update,
-                            connection_table1,
-                            stream_exit,
-                            stats,
-                            stake,
-                        ));
-                    } else {
-                        stats.connection_add_failed.fetch_add(1, Ordering::Relaxed);
-                    }
-                } else {
-                    connection.close(0u32.into(), &[0u8]);
-                    stats
-                        .connection_add_failed_unstaked_node
-                        .fetch_add(1, Ordering::Relaxed);
-                }
-            } else {
-                stats
-                    .connection_setup_timeout
-                    .fetch_add(1, Ordering::Relaxed);
-            }
+            tokio::spawn(connection_handshake(
```

Comment:
Any thoughts on how long to sleep?

Reply:
I guess, it comes down to how many connections we want to support per second. If we want 10k connections/sec, the sleep would be 100 microsec. If that sounds reasonable, we can do that for now.

Reply:
If the cap is 2500 connections per quic server, then 10k / sec would mean that connections could get pruned in roughly 250ms which is too short of a time window and it would be even worse for unstaked connections. How about we start with a sleep of 1ms for 1k connections per second?

Reply:
I was looking at it more from the TPS we want to support. If each client is unique, and we only support 1k connections, we will be throttling the TPS to 1K. Maybe that's an extreme case..

Reply:
I don't think it's correct to look at it that way. Most connections should be initiated by RPC nodes, bots, and other validators and be long lived with high throughput. One transaction per connection would be really inefficient and impractical

Reply:
Yes, that would make sense. I can change it to 1ms for now. I wish there was a way to dynamically adjust it somehow, in case we find 1ms to be too high a sleep.

I'll update the PR.

Reply:
Updated to 1000 microsec. Leaving it as micro seconds, in case we want to fine tune it later.